src/back/space/src/api/invitation.cpp

```diff
@@ -0,0 +1,336 @@
+#include "invitation.hpp"
+#include "../service/service.hpp"
+#include "../../../shared/headers.hpp"
+
+namespace svetit::space::handlers {
+
+Invitation::Invitation(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Invitation::GetList(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body) const
+{
+	formats::json::ValueBuilder res;
+
+	const auto& start = req.GetArg("start");
+	const auto& limit = req.GetArg("limit");
+	int iStart;
+	int iLimit;
+
+	if (!start.empty() && !limit.empty()){
+		try {
+			iStart = boost::lexical_cast<int>(start);
+		} catch(const std::exception& e) {
+			req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+			res["err"] = "Wrong start query param";
+			return res.ExtractValue();
+		}
+
+		try {
+			iLimit = boost::lexical_cast<int>(limit);
+		} catch(const std::exception& e) {
+			req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+			res["err"] = "Wrong limit query param";
+			return res.ExtractValue();
+		}
+
+		if (iStart < 0) {
+			req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+			res["err"] = "Start param must be valid";
+			return res.ExtractValue();
+		}
+
+		if (iLimit <= 0) {
+			req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+			res["err"] = "Limit param must be valid";
+			return res.ExtractValue();
+		}
+	} else {
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = "Start and Limit params must be set";
+		return res.ExtractValue();
+	}
+
+	try {
+		res["list"] = _s.GetInvitationList(iStart, iLimit);
+		res["total"] = _s.GetInvitationsCount();
+	}
+	catch(const std::exception& e) {
+		LOG_WARNING() << " Fail to get invitations list: " << e.what();
+		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
+		res["err"] = "Fail to get invitations list";
+	}
```

Comment:
Где-то в shared:
```c++
struct BadRequestException : public std::runtime_error {
	using std::runtime_error::runtime_error;
};

struct Paging {
	int start, limit;
};

Paging parsePaging(const server::http::HttpRequest& req)
{
	try {
		Paging info{
			.start = std::stoi(req.GetArg("start")),
			.limit = std::stoi(req.GetArg("limit"))
		};
		
		if (start < 0 || limit < 0)
			throw BadRequestException("range params less then zero");
		return info;
	} catch(const std::exception& e) {
		throw BadRequestException(e.what());
	}
	return {};
}
```

Везде в подобных запросах:
```suggestion
	try {
		auto paging = parsePaging(req);
		res["list"] = _s.GetInvitationList(paging.start, paging.limit);
		res["total"] = _s.GetInvitationsCount();
	}
	catch(const BadRequestException& e) {
		res["err"] = e.what();
		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
		return res.ExtractValue();
	}
	catch(const std::exception& e) {
		LOG_WARNING() << "Fail to get invitations list: " << e.what();
		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
		res["err"] = "Fail to get invitations list";
	}
```

Reply:
Именно в этом хендлере переписал, надо подумать, где это внедрить в других местах.

Reply:
Переписал парсинг параметров start и limit в остальных хендлерах. Осталось везде использовать исключение BadRequestException из shared.

Reply:
Готово.